snuba/datasets/factory.py

```diff
@@ -30,7 +33,7 @@ def get_dataset(name):
     from snuba.datasets.transactions import TransactionsDataset
     from snuba.datasets.outcomes import OutcomesDataset
 
-    dataset_mappings = {
+    dataset_factories: MutableMapping[str, Callable[[], Dataset]] = {
```

Comment:
The base `Dataset` constructor has required parameters that are passed by the parameterless subclasses, so `Type[Dataset]` won't work here (since you can't call `Dataset()` successfully.)